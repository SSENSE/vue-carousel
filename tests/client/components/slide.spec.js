const Vue = require('vue');
const utils = require('../utils');

const Carousel = require('../../../src/Carousel.vue');
const Slide = require('../../../src/Slide.vue');

describe('Slide', () => {
  it('should mount successfully', () => {
    const vm = new Vue({
      el: document.createElement('div'),
      render: (h) => h(Carousel, {}, [h(Slide)]),
    });
    const carouselInstance = vm.$children[0];
    const slideInstance = carouselInstance.$children[0];
    expect(slideInstance._isMounted).toBe(true);

    return utils.expectToMatchSnapshot(vm);
  });
});
