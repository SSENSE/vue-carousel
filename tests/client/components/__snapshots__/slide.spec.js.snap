// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should mount successfully 1`] = `
".VueCarousel-wrapper
  .VueCarousel-inner(role='listbox', style='visibility: hidden;')
    .VueCarousel-slide.VueCarousel-slide-active(tabindex='-1')
|  
.VueCarousel-pagination(style='display: none;')
  ul.VueCarousel-dot-container(role='tablist')
    li.VueCarousel-dot.VueCarousel-dot--active(aria-hidden='false', role='presentation', aria-selected='true', style='margin-top: 20px; padding: 10px;')
      button.VueCarousel-dot-button(type='button', role='button', tabindex='0', style='width: 10px; height: 10px; background: rgb(0, 0, 0);')
// 
"
`;
